src/EnergyPlus/PlantLoopEquip.cc

```diff
@@ -308,19 +307,19 @@ namespace PlantLoopEquip {
 		//Pipe has no special types at the moment, so find it this way
 		if ( GeneralEquipType == GenEquipTypes_Pipe ) {
 			if ( EquipTypeNum == TypeOf_Pipe ) {
```

Comment:
I guess i'm confused why you need the conditionals as well. Why can't you just have one call:

`sim_component.compPtr->simulate( sim_component_location, FirstHVACIteration, InitLoopEquip );`